src/Image_Backend.php

```diff
@@ -84,87 +78,61 @@ public function setImageResource($resource);
-     *
-     * @param int $width
-     * @param int $height
-     * @param bool $useAsMinimum If true, image will be sized outside of these dimensions.
-     * If false (default) image will be sized inside these dimensions.
-     * @return static
+     * Resize the image by preserving aspect ratio. By default, the image cannot be resized to be larger
+     * than its current size.
+     * Passing true to useAsMinimum will allow the image to be scaled up.
      */
-    public function resizeRatio($width, $height, $useAsMinimum = false);
+    public function resizeRatio(int $width, int $height, bool $useAsMinimum = false): static;
 
     /**
      * Resize an image by width. Preserves aspect ratio.
-     *
-     * @param int $width
-     * @return static
      */
-    public function resizeByWidth($width);
+    public function resizeByWidth(int $width): static;
 
     /**
      * Resize an image by height. Preserves aspect ratio.
-     *
-     * @param int $height
-     * @return static
      */
-    public function resizeByHeight($height);
+    public function resizeByHeight(int $height): static;
 
     /**
-     * Return a clone of this image resized, with space filled in with the given colour
-     *
-     * @param int $width
-     * @param int $height
-     * @param string $backgroundColor
-     * @param int $transparencyPercent
-     * @return static
+     * Return a clone of this image resized, with space filled in with the given colour.
      */
-    public function paddedResize($width, $height, $backgroundColor = "FFFFFF", $transparencyPercent = 0);
+    public function paddedResize(string $width, string $height, string $backgroundColour = 'FFFFFF', int $transparencyPercent = 0): static;
 
     /**
      * Resize an image to cover the given width/height completely, and crop off any overhanging edges.
-     *
-     * @param int $width
-     * @param int $height
-     * @return static
      */
-    public function croppedResize($width, $height);
+    public function croppedResize(int $width, int $height, string $position): static;
 
     /**
      * Crop's part of image.
-     * @param int $top y position of left upper corner of crop rectangle
-     * @param int $left x position of left upper corner of crop rectangle
+     * @param int $top Amount of pixels the cutout will be moved on the y (vertical) axis
+     * @param int $left Amount of pixels the cutout will be moved on the x (horizontal) axis
      * @param int $width rectangle width
      * @param int $height rectangle height
-     * @return Image_Backend
+     * @param string $position Postion at which the cutout will be aligned
+     * @param string $backgroundColour Colour to fill any newly created areas
      */
-    public function crop($top, $left, $width, $height);
+    public function crop(int $top, int $left, int $width, int $height, string $position, string $backgroundColour = 'FFFFFF'): static;
```

Comment:
Added `$position` and `$backgroundColour` arguments for advanced control, should it be needed.